### DISCUSSION TITLE
N2K devices not seen on Halmet

### DISCUSSION THREAD

**Discussion by big-col** [hatlabs/discussions#123] [General]
I have a newly fitted Airmar DST810 that has both bluetooth and n2k connectivity.   I can seem to read anything on the n2k bus, the RX\TX leds are not blinking.  see photos of the 'on the floor' installation before it all gets fitted in place.  Is there and config required in the main.cpp file to enable this?

The 3 connections are for
1) Separate power - as i cant find the connection to go into the housing
2) Data connection to Halmet
3) Transducer

![WIN_20240519_13_24_25_Pro](https://github.com/hatlabs/discussions/assets/10856730/5e900911-b66f-4fb5-8819-b24cb00ce2a0)
![WIN_20240519_13_24_11_Pro](https://github.com/hatlabs/discussions/assets/10856730/9c7fe068-6511-4ca6-8515-b2082c20daa9)

**Comment by big-col** [comment]
Excuse the mess!!

**Comment by mairas** [comment]
For a while I thought someone had chomped a piece off from that poor Victron device, but then I realized it was a mid-air HALMET. :-D

It's not quite clear to me what's happening. Is this a new "mini" N2K network? And the devices connected to the T-connector are  DST810, HALMET and power input? What's happening on the HALMET N2K connector? It's powered from somewhere else and has just the H/L pair connected from the N2K backbone?

**Reply by big-col** [reply]
You are quite right in all of your assumptions. For now there is a separate power feed for the Halmet, it will come from the N2K network next week..

**Reply by mairas** [reply]
But the N2K network is still powered, right?

What kind of software are you running on the HALMET? The NMEA2000 device needs to be initialized before the LEDs do anything sensible, IIRC.

**Reply by big-col** [reply]
It is powered,.yes. power, connection to Halmet, comment to Transducer are the 3 connections you see....

If I drop my main.cpp file here would that give you enough info to see if I've something missing? I was thinking I've not Included everything needed....

**Reply by mairas** [reply]
Yep, that'd be enough.

**Comment by big-col** [comment]
[colin.main.txt](https://github.com/hatlabs/discussions/files/15370386/colin.main.txt)

**Reply by mairas** [reply]
Code looks fine. Are the HALMET power and the N2K connected to the same ground?

**Reply by big-col** [reply]
Yes, same ground

**Reply by big-col** [reply]
> Yes, same ground

Is this the right answer?

**Reply by mairas** [reply]
Yes, sure. If it doesn't work even after you fix the wiring, first check the continuity of the N2K connections with a multimeter to verify that the L and H signals actually reach the HALMET device.

**Reply by big-col** [reply]
I'm reading just over 2v across the blue/white connections.  I've also downloaded the sample config from your site and installed on a spare Halmet. Same issue. The N2K green LEDs are lit but not flashing. 


...I'm not seeing anything in SK, do I need any other plugins? 

Thanks
Colin

**Reply by big-col** [reply]
@mairas no continuity across the data lines....

**Reply by big-col** [reply]
A new drop wire and it seems to be working 🥳🥳

**Reply by big-col** [reply]
Although the N2K LEDs are flickering nicely, I am still not getting N2K data into SK. The connection is established...

**Reply by mairas** [reply]
So far the question has been about HALMET and NMEA 2000. What data are you transmitting on HALMET? What is your Signal K hardware setup like? Are you receiving other N2K traffic successfully?

**Reply by big-col** [reply]
i have a much clearer and updated disucssion on Discord - would you like to move this discussion to there or carry on here?   im happy either way - never quite sure which forum to use :)
